lib/licensed/sources/bundler.rb

```diff
@@ -69,7 +100,7 @@ def recursive_specs(specs, results = Set.new)
       def specs_for_dependencies(dependencies, source)
         included_dependencies = dependencies.select { |d| include?(d, source) }
         included_dependencies.map do |dep|
-          gem_spec(dep) || raise("Unable to find a specification for #{dep.name} (#{dep.requirement}) in any sources")
+          gem_spec(dep) || MissingSpecification.new(name: dep.name, requirement: dep.requirement)
```

Comment:
Creating a new "error" specification type seemed to be the easiest way to capture gem specifications that failed, as it fits into the rest of the source evaluation pretty cleanly